experimental/Schemes/Sheaves.jl

```diff
@@ -164,6 +164,174 @@ end
 # The implementation of StructureSheafOfRings                         #
 ########################################################################
 
+function _is_open_func_for_schemes(X::AbsCoveredScheme)
+  ### Checks for open containment.
+  #
+  # We allow the following cases:
+  #
+  #  * U::PrincipalOpenSubset with one ancestor W in the basic charts of X
+  #  * U::SimplifiedSpec with one ancestor W in the basic charts of X
+  #  * U::PrincipalOpenSubset ⊂ V::PrincipalOpenSubset with ambient_scheme(U) === ambient_scheme(V) in the basic charts of X
+  #  * U::PrincipalOpenSubset ⊂ V::PrincipalOpenSubset with ambient_scheme(U) != ambient_scheme(V) both in the basic charts of X
+  #    and U and V contained in the glueing domains of their ambient schemes
+  #  * U::AbsSpec ⊂ U::AbsSpec in the basic charts of X
+  #  * U::AbsSpec ⊂ X for U in the basic charts
+  #  * U::PrincipalOpenSubset ⊂ X with ambient_scheme(U) in the basic charts of X
+  #  * W::SpecOpen ⊂ X with ambient_scheme(U) in the basic charts of X
+  function is_open_func(
+      U::Union{<:PrincipalOpenSubset, <:SimplifiedSpec}, 
+      V::Union{<:PrincipalOpenSubset, <:SimplifiedSpec}
+    )
+    inc_U_flat = _flatten_open_subscheme(U, default_covering(X))
+    inc_V_flat = _flatten_open_subscheme(V, default_covering(X))
+    A = ambient_scheme(codomain(inc_U_flat))
+    B = ambient_scheme(codomain(inc_V_flat))
+    Udirect = codomain(inc_U_flat)
+    Vdirect = codomain(inc_V_flat)
+   # some_ancestor(W->any(WW->(WW===W), affine_charts(X)), U) || return false
+   # some_ancestor(W->any(WW->(WW===W), affine_charts(X)), V) || return false
+   # incU, dU = _find_chart(U, default_covering(X))
+   # incV, dU = _find_chart(V, default_covering(X))
+   # A = codomain(incU)
+   # B = codomain(incV)
+   # Udirect = (ambient_scheme(U) === A ? U : PrincipalOpenSubset(codomain(incU), dU))
+   # Vdirect = (ambient_scheme(V) === B ? V : PrincipalOpenSubset(codomain(incV), dV))
+
+    if A === B
+      is_subset(Udirect, Vdirect) || return false
+    else
+      G = default_covering(X)[A, B] # Get the glueing
+      f, g = glueing_morphisms(G)
+      is_subset(Udirect, domain(f)) || return false
+      gU = preimage(g, Udirect)
+      is_subset(gU, Vdirect) || return false
+    end
+    return true
+  end
+  function is_open_func(
+      U::Union{<:PrincipalOpenSubset, <:SimplifiedSpec}, 
+      Y::AbsCoveredScheme
+    )
+    return Y === X && some_ancestor(W->(W in affine_charts(X)), U)
+  end
+  function is_open_func(U::AbsSpec, Y::AbsCoveredScheme)
+    return Y === X && some_ancestor(W->any(WW->(WW===W), affine_charts(X)), U)
+  end
+  function is_open_func(Z::AbsCoveredScheme, Y::AbsCoveredScheme)
```

Comment:
I'm afraid I do not fully get what you have in mind here. You are speaking about line 220?

Reply:
I am talking about the function beginning in 220 and only doing the Check X===Y===Z.
So is_open_func is designed to only return true for two AbsCoveredSchemes, if the AbsCoveredSchemes are equal. This is by choice and this is fine, but it could happen that the user expects that it also works if Y and Z are subschemes of X by selection of subsets of the patches -- if e.g. X is the disjoint union of U_1,...,U_4,  and the other two are the disjoint unions of U_1,U_2 and of U_1,...,U_3 respectively.